slsim/Sources/SupernovaeCatalog/supernovae_sample.py

```diff
@@ -30,27 +33,55 @@ def supernovae_host_galaxy_offset(host_galaxy_catalog):
 
     offsets = []
     position_angle = []
+    e1 = []
+    e2 = []
 
     for i in range(len(host_galaxy_catalog)):
+
+        # Set a limit on maximum SN Ia offset ratio from host galaxy center
         while offset_ratios[i] > 3:
             offset_ratios[i] = stats.lognorm.rvs(
                 0.764609, loc=-0.0284546, scale=0.450885, size=1
             )[0]
 
+        # Calculate offsets [rad]
         offsets.append(offset_ratios[i] * list(host_galaxy_catalog["angular_size"])[i])
+
+        # Generate random angle
         position_angle.append(np.random.uniform(0, 360))
 
+        # Calculate projected eccentricities
+        temp_e1, temp_e2 = galaxy_projected_eccentricity(
+            host_galaxy_catalog["ellipticity"][i]
+        )
+        e1.append(temp_e1)
+        e2.append(temp_e2)
+
+    # Calculate the ra and dec coordinates of the offset [arcsec]
     host_center = SkyCoord(1 * units.deg, 1 * units.deg, frame="icrs")
     offsets = host_center.directional_offset_by(
         position_angle * units.deg, offsets * units.rad
     )
+    original_ra_off = (offsets.ra - 1 * units.deg).to(units.arcsec)
+    original_dec_off = (offsets.dec - 1 * units.deg).to(units.arcsec)
 
-    ra_off = offsets.ra - 1 * units.deg
-    ra_off = ra_off.to(units.arcsec)
-    dec_off = offsets.dec - 1 * units.deg
-    dec_off = dec_off.to(units.arcsec)
+    transformed_ra_off = []
+    transformed_dec_off = []
+
+    # Transform the offset coordinates with eccentricities e1, e2 into elliptical coordinate system
+    for i in range(len(host_galaxy_catalog)):
+        ra_off, dec_off = transform_e1e2_product_average(
+            original_ra_off[i],
+            original_dec_off[i],
+            e1[i],
```

Comment:
For slsim to lenstronomy conversion of ellipticities, use this function: https://github.com/LSST-strong-lensing/slsim/blob/c6b3fa82eae8a0e58e7ec539d69ce328a7a30689/slsim/Util/param_util.py#L261 .